include/zephyr/drivers/wifi/nrfwifi/off_raw_tx/off_raw_tx_api.h

```diff
@@ -0,0 +1,258 @@
+	/** 9 Mbps */
+	RATE_9M,
+	/** 12 Mbps */
+	RATE_12M,
+	/** 18 Mbps */
+	RATE_18M,
+	/** 24 Mbps */
+	RATE_24M,
+	/** 36 Mbps */
+	RATE_36M,
+	/** 48 Mbps */
+	RATE_48M,
+	/** 54 Mbps */
+	RATE_54M,
+	/** MCS 0 */
+	RATE_MCS0,
+	/** MCS 1 */
+	RATE_MCS1,
+	/** MCS 2 */
+	RATE_MCS2,
+	/** MCS 3 */
+	RATE_MCS3,
+	/** MCS 4 */
+	RATE_MCS4,
+	/** MCS 5 */
+	RATE_MCS5,
+	/** MCS 6 */
+	RATE_MCS6,
+	/** MCS 7 */
+	RATE_MCS7,
+	/** Invalid rate */
+	RATE_MAX
+};
+
+
+/**
+ * @brief- HE guard interval value
+ * Value of the guard interval to be used between symbols when transmitting using HE.
+ */
+enum nrf_wifi_off_raw_tx_he_gi {
+	/** 800 ns */
+	HE_GI_800NS,
+	/** 1600 ns */
+	HE_GI_1600NS,
+	/** 3200 ns */
+	HE_GI_3200NS,
+	/** Invalid value */
+	HE_GI_MAX
+};
+
+
+/**
+ * @brief- HE long training field duration
+ * Value of the long training field duration to be used when transmitting using HE.
+ */
+enum nrf_wifi_off_raw_tx_he_ltf {
+	/** 3.2us */
+	HE_LTF_3200NS,
+	/** 6.4us */
+	HE_LTF_6400NS,
+	/** 12.8us */
+	HE_LTF_12800NS,
+	/** Invalid value */
+	HE_LTF_MAX
+};
+
+/**
+ * @brief- Throughput mode
+ * Throughput mode to be used for transmitting the packet.
+ */
+enum nrf_wifi_off_raw_tx_tput_mode {
+	/** Legacy mode */
+	TPUT_MODE_LEGACY,
+	/** High Throughput mode (11n) */
+	TPUT_MODE_HT,
+	/** Very high throughput mode (11ac) */
+	TPUT_MODE_VHT,
+	/** HE SU mode */
+	TPUT_MODE_HE_SU,
+	/** HE ER SU mode */
+	TPUT_MODE_HE_ER_SU,
+	/** HE TB mode */
+	TPUT_MODE_HE_TB,
+	/** Highest throughput mode currently defined */
+	TPUT_MODE_MAX
+};
+
+/**
+ * @brief This structure defines the Offloaded raw tx debug statistics.
+ *
+ */
+struct nrf_wifi_off_raw_tx_stats {
+	/** Number of packets sent */
+	unsigned int off_raw_tx_pkt_sent;
+};
+
+/**
+ * @brief- Configuration parameters for offloaded raw TX
+ * Parameters which can be used to configure the offloaded raw TX operation.
+ */
+struct nrf_wifi_off_raw_tx_conf {
+	/** Time interval (in microseconds) between transmissions */
+	unsigned int period_us;
+	/** Transmit power in dBm (0 to 20) */
+	unsigned int tx_pwr;
+	/** Channel number on which to transmit */
+	unsigned int chan;
+	/** Set to TRUE to use short preamble for FALSE to disable short preamble */
+	bool short_preamble;
+	/* Number of times a packet should be retried at each possible rate */
+	unsigned int num_retries;
+	/** Throughput mode for packet transmittion. Refer &enum nrf_wifi_off_raw_tx_tput_mode */
+	enum nrf_wifi_off_raw_tx_tput_mode tput_mode;
+	/* Rate at which packet needs to be transmitted. Refer &enum nrf_wifi_off_raw_tx_rate */
+	enum nrf_wifi_off_raw_tx_rate rate;
+	/** HE GI. Refer &enum nrf_wifi_off_raw_tx_he_gi */
+	enum nrf_wifi_off_raw_tx_he_gi he_gi;
+	/** HE GI. Refer &enum nrf_wifi_off_raw_tx_he_ltf */
+	enum nrf_wifi_off_raw_tx_he_ltf he_ltf;
+	/* Pointer to packet to be transmitted */
+	void *pkt;
+	/** Packet length of the frame to be transmitted, (min 26 bytes and max 600 bytes) */
+	unsigned int pkt_len;
```

Comment:
better to swap these to keep the variable part last, also `pkt[]` might be better.

Reply:
From the sample we are passing array to the pkt pointer, it will have the location of the array and directly copy that array to firmware.

Reply:
Sure, you can do that with the swapped members as well, would be useful in case you want to attach the payload at the end.